daemon/daemon.go

```diff
@@ -939,11 +939,11 @@ func NewDaemon(ctx context.Context, config *config.Config, pluginStore *plugin.S
 	// the graphdrivers it should be safe to do here.
 	_, doStorageMigration := os.LookupEnv("BALENA_MIGRATE_OVERLAY")
 	if config.GraphDriver == "overlay2" && doStorageMigration {
-		if storagemigration.CheckAufsRootExists(config.Root) != nil &&
-			storagemigration.CheckOverlayRootExists(config.Root) != nil {
+		if storagemigration.CheckAufsRootExists(config.Root) == nil &&
```

Comment:
This is a big improvement for sure (like, now we are doing something that makes sense). I'd be OK to merge it as is, but here something for consideration:

I don't like very much the design of `Check{Aufs,Overlay}RootExists()` because a non-nil return value can mean both "root does not exist" or "a real error happened" depending on what is the exact `error` value. I think the resulting client code is kinda confusing because of that. Maybe we would even have spotted this very issue before if these functions returned `(bool, error)`.

Anyway, even if we decide to not change `Check{Aufs,Overlay}RootExists()`, it would probably be nice to check and handle the real errors (that is, the cases in which the return value is neither `nil` nor `Err{Aufs,Overlay}RootNotExists`).

Reply:
agreed. looking over the code, it appears that we only call those helpers from `daemon.go` anyway...